src/rdf4cpp/rdf/node/IRIResource.cpp

```diff
@@ -1,5 +1,20 @@
-//
-// Created by Lixi Alié Conrads on 5/30/21.
-//
-
 #include "IRIResource.h"
+
+namespace rdf4cpp::rdf::node {
+
+IRIResource::IRIResource(void *ptr, const RDFNode::NodeID &id) : RDFNode(ptr, id) {}
+IRIResource::IRIResource() : RDFNode() {}
+IRIResource::IRIResource(const std::string &iri, RDFNode::NodeManager &node_manager)
```

Comment:
just as a note: As discussed in final we will use IRI object, rather than string to assure validity